.gitlab/.gitlab-ci.yml

```
@@ -115,17 +121,36 @@ charon-build_conda_env:
 .build_4C:
   stage: build
   script:
-    - ssh queens@$REMOTE_ADDRESS "$LOAD_MODULES cd ~/workspace; rm -rf fourc_source;
-      rm -rf fourc_build; git clone https://github.com/4C-multiphysics/4C.git fourc_source;
-      mkdir fourc_build; cd fourc_build; cmake --preset=$PRESET_4C ../fourc_source; ninja -j 8 full"
+    - |
+      COMMAND="bash -lc 'srun \
+        --job-name="queens_ci_build_4c_$CI_PIPELINE_ID" \
+        --nodes=1 \
+        --ntasks-per-node=1 \
+        --cpus-per-task=$SLURM_CPUS \
+        --time=$SLURM_TIME \
+        $SLURM_EXTRA_ARGUMENTS \
+        bash -c \" \
+          $LOAD_MODULES \
+          module list ; \
+          cd ~/workspace ; \
+          rm -rfv fourc_source fourc_build ; \
+          git clone --depth 1 https://github.com/4C-multiphysics/4C.git fourc_source ; \
+          mkdir fourc_build ; \
+          cd fourc_build ; \
+          cmake --preset=$PRESET_4C ../fourc_source ; \
+          ninja -j $SLURM_CPUS full \"'"
+    - echo $COMMAND
+    - ssh queens@$REMOTE_ADDRESS $COMMAND
 
 thought-build_4C:
   extends:
     - .build_4C
   variables:
     LOAD_MODULES: ""
-    REMOTE_ADDRESS: "thought"
+    REMOTE_ADDRESS: $LNM_REMOTE_ADDRESS
     PRESET_4C: "lnm_thought"
```

```suggestion
    PRESET_4C: "lnm_thought -DFOUR_C_ENABLE_METADATA_GENERATION=OFF"
```
For the metadata generation 4C is executed, which seems to cause an MPI error here. As we don't need the metadata here, we can disable its generation. Thanks to @davidrudlstorfer !

---

maybe it would make sense to introduce a global variable with `ADDITIONAL_CMAKE_VARIABLES`. These could be reused in other scenarios and it would probably also make sense to disable the metadata generation for the IMCS job on charon because these files are not needed at all.

---

The metadata generation is already disabled within the standard `imcs_charon` preset.  I am wondering why this error only occurs here and not within your other general setup/testing. Is there something special with this environment or @davidrudlstorfer would be a better solution to simply update the preset `lnm_thought` within 4c? 
Of course queens can start overwriting/ deploying their own specialized 4c version for testing, but I am not quite sure if this maintenance overhead is really necessary due to one setting or are there more specific settings you need to turn on/off explicitly for the 4c queens version?

---

@knarfnitram thanks for your input. Yes it makes total sense to just update the cluster presets for `thought` and `bruteforce` in 4C and disable the metadata generation completely.

---

Solved with https://github.com/4C-multiphysics/4C/pull/778